app/index.js

```diff
@@ -51,6 +52,16 @@ module.exports = function App(config) {
 
   app.use(bodyParser.json());
 
+  app.middlewares = [];
+
+  app.use((req, res, next) => {
+    async.series(app.middlewares.map(middleware => cb => middleware(req, res, cb)), err => next(err));
```

Comment:
So basically, we have a single Express middleware executing all consumer configured "middlewares". This line will make my future self's head hurt, a comment describing purpose here would provide clarity.

Reply:
@ryanricard I'll add a comment. Sorry it's sort of confusing. It's a micro-reimplementation of the middleware flow that Express has internally. Technically I'm not handling error middleware, but that may be an edge case. I'll see if I can think of an easy way to do so, or even look for or make a 3rd party module to do it.

Reply:
@ryanricard Looks like this would work: https://github.com/blakeembrey/compose-middleware